compiler/rustc_middle/src/mir/interpret/allocation/init_mask.rs

```diff
@@ -182,11 +184,33 @@ impl InitMask {
 /// The actual materialized blocks of the bitmask, when we can't keep the `InitMask` lazy.
 // Note: for performance reasons when interning, some of the fields can be partially
 // hashed. (see the `Hash` impl below for more details), so the impl is not derived.
-#[derive(Clone, Debug, Eq, PartialEq, TyEncodable, TyDecodable, HashStable)]
+#[derive(Clone, Debug, Eq, PartialEq, HashStable)]
 struct InitMaskMaterialized {
     blocks: Vec<Block>,
 }
 
+impl<E: TyEncoder> Encodable<E> for InitMaskMaterialized {
```

Comment:
Should `encode` and `decode` each have a brief comment explaining why they don't use `derive`?

Reply:
I'd put a comment on the decode impl that just refers to the encode impl :)

Reply:
Done